quinn/src/tests.rs

```diff
@@ -407,8 +418,9 @@ fn run_echo(client_addr: SocketAddr, server_addr: SocketAddr) {
             assert_eq!(&data[..], b"foo");
             new_conn.connection.close(0u32.into(), b"done");
         });
-    }
-    runtime.run().unwrap();
+        handler
+    };
+    runtime.block_on(handler).unwrap();
```

Comment:
This test should cover clean shutdown by blocking on *all* spawned tasks.

Reply:
Done.